ansible/roles/satellite-register-host/tasks/register.yml

```diff
@@ -19,6 +19,19 @@
   package:
     name: katello-ca-consumer-*.noarch
     state: absent
+  when:
+   - cloud_provider != 'azure'
+   - cloud_provider != 'gcp'
```

Comment:
I don't understand the link between satellite-register-host  and the cloud provider.

Can you explain ?

Reply:
The images used as part of the Azure and GCP marketplace offerings have no Satellite related packages installed currently. 
Since this could change in the future, I've excluded them from the mandatory remove and adds and optional one right below it, so at some point if the image starts to include Satellite it will still get cleaned up.
Not ideal, but if we want to consistently use Satellite I don't know another way without having yet more images to maintain.

Reply:
But if the package is not there, then nothing will be done, right ?

IMO those condition are not mandatory and create confusion.